exchanges/huobi/huobi_websocket.go

```diff
@@ -75,427 +73,240 @@ var subscriptionNames = map[string]string{
+	subscription.MyOrdersChannel:  wsMyOrdersChannel,
 	subscription.MyAccountChannel: wsMyAccountChannel,
-	*/
 }
 
-// Instantiates a communications channel between websocket connections
-var comms = make(chan WsMessage)
-
 // WsConnect initiates a new websocket connection
 func (h *HUOBI) WsConnect() error {
 	if !h.Websocket.IsEnabled() || !h.IsEnabled() {
 		return stream.ErrWebsocketNotEnabled
 	}
-	var dialer websocket.Dialer
-	err := h.wsDial(&dialer)
-	if err != nil {
+	if err := h.Websocket.Conn.Dial(&websocket.Dialer{}, http.Header{}); err != nil {
 		return err
 	}
+	ctx := context.Background()
 
-	if h.Websocket.CanUseAuthenticatedEndpoints() {
-		err = h.wsAuthenticatedDial(&dialer)
-		if err != nil {
-			log.Errorf(log.ExchangeSys,
-				"%v - authenticated dial failed: %v\n",
-				h.Name,
-				err)
-		}
-		err = h.wsLogin(context.TODO())
-		if err != nil {
-			log.Errorf(log.ExchangeSys,
-				"%v - authentication failed: %v\n",
-				h.Name,
-				err)
-			h.Websocket.SetCanUseAuthenticatedEndpoints(false)
-		}
-	}
-
-	h.Websocket.Wg.Add(1)
-	go h.wsReadData()
-	return nil
-}
-
-func (h *HUOBI) wsDial(dialer *websocket.Dialer) error {
-	err := h.Websocket.Conn.Dial(dialer, http.Header{})
-	if err != nil {
-		return err
-	}
-	h.Websocket.Wg.Add(1)
-	go h.wsFunnelConnectionData(h.Websocket.Conn, wsMarketURL)
-	return nil
-}
+	ch := make(chan []byte)
+	h.Websocket.Wg.Add(2)
+	go h.wsReadMsgs(ch)
+	go h.wsFunnelMsgs(h.Websocket.Conn, ch)
 
-func (h *HUOBI) wsAuthenticatedDial(dialer *websocket.Dialer) error {
-	if !h.IsWebsocketAuthenticationSupported() {
-		return fmt.Errorf("%v AuthenticatedWebsocketAPISupport not enabled",
-			h.Name)
-	}
-	err := h.Websocket.AuthConn.Dial(dialer, http.Header{})
-	if err != nil {
-		return err
+	h.Websocket.SetCanUseAuthenticatedEndpoints(false)
+	if h.IsWebsocketAuthenticationSupported() {
+		if err := h.wsAuthConnect(ctx); err != nil {
+			return fmt.Errorf("error authenticating websocket: %w", err)
+		}
+		h.Websocket.Wg.Add(1)
+		go h.wsFunnelMsgs(h.Websocket.AuthConn, ch)
 	}
 
-	h.Websocket.Wg.Add(1)
-	go h.wsFunnelConnectionData(h.Websocket.AuthConn, wsAccountsOrdersURL)
 	return nil
 }
 
-// wsFunnelConnectionData manages data from multiple endpoints and passes it to
-// a channel
-func (h *HUOBI) wsFunnelConnectionData(ws stream.Connection, url string) {
+// wsFunnelMsgs relays messages from a websocket to a channel for central processing
+func (h *HUOBI) wsFunnelMsgs(s stream.Connection, ch chan []byte) {
 	defer h.Websocket.Wg.Done()
 	for {
-		resp := ws.ReadMessage()
-		if resp.Raw == nil {
+		msg := s.ReadMessage()
+		if msg.Raw == nil {
 			return
 		}
-		comms <- WsMessage{Raw: resp.Raw, URL: url}
+		ch <- msg.Raw
 	}
 }
 
-// wsReadData receives and passes on websocket messages for processing
-func (h *HUOBI) wsReadData() {
+// wsReadMsgs receives messages from a message funnel and processes them
+func (h *HUOBI) wsReadMsgs(ch chan []byte) {
```

Comment:
Can remove funnelling and channel

```diff
diff --git a/exchanges/huobi/huobi_websocket.go b/exchanges/huobi/huobi_websocket.go
index 269d72e46..0acd854c9 100644
--- a/exchanges/huobi/huobi_websocket.go
+++ b/exchanges/huobi/huobi_websocket.go
@@ -88,10 +88,8 @@ func (h *HUOBI) WsConnect() error {
        }
        ctx := context.Background()

-       ch := make(chan []byte)
-       h.Websocket.Wg.Add(2)
-       go h.wsReadMsgs(ch)
-       go h.wsFunnelMsgs(h.Websocket.Conn, ch)
+       h.Websocket.Wg.Add(1)
+       go h.wsReadMsgs(h.Websocket.Conn)

        h.Websocket.SetCanUseAuthenticatedEndpoints(false)
        if h.IsWebsocketAuthenticationSupported() {
@@ -99,38 +97,26 @@ func (h *HUOBI) WsConnect() error {
                        return fmt.Errorf("error authenticating websocket: %w", err)
                }
                h.Websocket.Wg.Add(1)
-               go h.wsFunnelMsgs(h.Websocket.AuthConn, ch)
+               go h.wsReadMsgs(h.Websocket.AuthConn)
        }

        return nil
 }

-// wsFunnelMsgs relays messages from a websocket to a channel for central processing
-func (h *HUOBI) wsFunnelMsgs(s stream.Connection, ch chan []byte) {
+// wsReadMsgs reads and processes data from a websocket connection
+func (h *HUOBI) wsReadMsgs(s stream.Connection) {
        defer h.Websocket.Wg.Done()
        for {
                msg := s.ReadMessage()
                if msg.Raw == nil {
                        return
                }
-               ch <- msg.Raw
-       }
-}
-
-// wsReadMsgs receives messages from a message funnel and processes them
-func (h *HUOBI) wsReadMsgs(ch chan []byte) {
-       defer h.Websocket.Wg.Done()
-       for {
-               select {
-               case <-h.Websocket.ShutdownC:
-                       return
-               case msg := <-ch:
-                       if err := h.wsHandleData(msg); err != nil {
-                               h.Websocket.DataHandler <- err
-                       }
+               if err := h.wsHandleData(msg.Raw); err != nil {
+                       h.Websocket.DataHandler <- err
                }
        }
 }
+
 func (h *HUOBI) wsHandleData(respRaw []byte) error {
        for _, op := range []string{wsSubOp, wsUnsubOp} {
                key := op + "bed" // subbed, unsubbed
```

Reply:
Agreed. Thanks for this! 🙏 

_Aside: I always think `data` is bad in nearly all contexts. In the context of this comment, we know it's a message.
At some point I want to go on a `respRaw` destroying crusade too. And `wsHandleData`. Is there a wsDoThings around as well?_

Fixed https://github.com/gbjk/gocryptotrader/commit/479095c99